src/tracer.js

```diff
@@ -237,7 +240,9 @@ export default class Tracer {
             ctx.baggage = parent.baggage;
 
             parent.finalizeSampling();
-            this.processDeferredSampling(ctx, operationName, internalTags);
+            if (this._upsampleChildSpans) {
+                this.upsampleChild(parent, ctx, operationName, internalTags);
+            }
             ctx.finalizeSampling();
```

Comment:
not sure this is correct anymore. When upsampling a child we still need to account for the possibility of operationName changing and us having to re-sample it